### DISCUSSION TITLE
Ruleset-specific Audio Offset, Show Converted toggle and Difficulty Range

### DISCUSSION THREAD

**Discussion by DGCK81LNN** [ppy/osu#27951] [Ideas]
I like lazer's harmony of different rulesets, the shared UI elements and stuff. But sometimes I need settings to be ruleset-specific.

  * I want -20 ms Audio Offset for *osu!standard only*.
  * I want converted beatmaps to be shown only for *third-party rulesets except Yoso*.
  * Based on my current skill, say, I want 4 ~ 5.5-star *osu!standard* maps, 2.5 ~ 3.5-star *osu!taiko* maps, 1.5 ~ 3-star *osu!catch* maps and 1 ~ 2-star *osu!mania* maps.

Having to change these settings when switching between rulesets makes me less eager to play different rulesets. It would be very nice if the game remembered my settings for each ruleset.

**Comment by DGCK81LNN** [comment]
Recently I started to feel the need for ruleset-specific Ignore Beatmap Hitsounds toggle and skin hitsounds too, like, ignore beatmap hitsounds for taiko, and silent keysounds for mania _by default_ (i.e. unless provided by beatmap)

It would also be nice if we had the Storyboard and Ignore Beatmap Skin/Hitsounds toggles remembered on a per-beatmap basis too.

**Comment by OCDkirby** [comment]
+1 for gamemode-specific offset. I'm assuming it has to do with different map engines and number of objects allocated/deallocated in the respective gamemodes, because I notice that my hits on osu!mania are consistently later than osu!standard (this would track with your wanting earlier offset on standard).

For now, the workaround is to set the audio offset for standard and fiddle with hitposition in my mania skin until that approximates the visual offset that I need (this creates a strange-looking gap between notes hitting the judgment line and the lane elements, but I look far enough up the lanes when playing that I can ignore it). This breaks if I change my scroll speed so I don't want to use it long-term

**Reply by DGCK81LNN** [reply]
Actually i suspect my problem is that my machine has a ~80 ms audio output latency or something. To play mania properly I'll have to set the audio offset to -80 ms and turn off keysounds, and it sucks so I've given up. I can get used to that delay on other gamemodes but I need -20 ms offset on standard or I still get too many 100s, perhaps due to my bad aiming.

**Reply by OCDkirby** [reply]
Ahh, I see. You might not be able to detect my issue then, I see about a 5ms difference between mania and standard on my global offset (+8 to +20ms, depending on setup)

Well-timed standard map, 0 local offset: 
![image](https://github.com/ppy/osu/assets/138795235/56a02ffb-4041-4b31-89c3-d0a826ad7b01)

Well-timed mania map, 0 local offset, scroll speed 29, hitposition/judgment line exactly where it should be for my skin: 
![image](https://github.com/ppy/osu/assets/138795235/978d71bb-55f8-4cce-922c-1f4c928de7a3)

What I feel would suggest that there's less latency on mania than standard (ie, my global offset is too early for mania). Usually if the issue is I'm unable to read mania patterns in time the hit error ends up being well over 10ms late (not to mention that the play that's 5ms behind feels like I'm reading and hitting everything on time), so this definitely isn't that.

**Reply by DGCK81LNN** [reply]
so recently I decided to try playing with sound effects off, and I discovered that I needed -70ms audio offset for mania but ~ -55ms for standard, the reverse of your situation. Not sure why. Perhaps I have a per-gamemode *output* offset in my head *lol* Also recently I figured -20ms was too much offset for stable and decided to set it at -5ms instead, which is barely noticable. Possibly something specific to my machine has changed, idk
